parsers/DK.py

```diff
@@ -43,13 +40,18 @@ def fetch_production(zone_key='DK-DK1', session=None,target_datetime=None,
                      
     
     url = 'https://api.energidataservice.dk/datastore_search_sql?sql={}'.format(sqlstr)
-    
     response = r.get(url)
-    
-    assert response.status_code == 200 and response.json()['result']['records'] != [], \
-        'Exception when fetching production for ' \
-        '{}: error when calling url={}'.format(zone_key, url)
-        
+
+    if response.status_code != 200:
+        error = response.json()['error']['__type']
+        text = response.json()['error']['info']['orig']
+        logger.warning('"{}" fetching production data for {}: {}'.format(
+            error, zone_key, text))
+        return
+    if response.json()['result']['records']:
+        logger.info('API returned no data for {} production'.format(zone_key))
```

Comment:
That's an error as well right? In that case, I would `raise`

Reply:
Hmmm.. I don't think no data being returned is an error. I see this sort of thing on a lot of data sources where it temporarily doesn't return data (maybe when they're updating a file or whatever) so one may see it a number of times during the day.. which would be annoying if you do get all the data eventually just not on every call to the API and as a result you've got all these not very useful log messages.
Just my two cents, happy to go with whatever!

Reply:
A parser should always return data. When a request for fetching data occurs, and that the parser is not able to deliver, then it should crash. This will ensure that the proper errors show up in kibana.
This is also helpful when collecting past data: it enables us to know when the data has been successfully retrieved.